tests/unit/directory_test.sh

```diff
@@ -0,0 +1,180 @@
+    "$(Console::printFailedTest \
+      "Assert directory exists should not work with files" "$a_file" "to exist but" "do not exist")"\
+    "$(assert_directory_exists "$a_file")"
+}
+
+function test_successful_assert_directory_not_exists() {
+  local a_directory="a_random_directory_that_will_not_exist"
+
+  assert_empty "$(assert_directory_not_exists "$a_directory")"
+}
+
+function test_unsuccessful_assert_directory_not_exists() {
+  local a_directory
+  a_directory="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+  assert_equals\
+    "$(Console::printFailedTest \
+      "Unsuccessful assert directory not exists" "$a_directory" "to not exist but" "the directory exists")"\
+    "$(assert_directory_not_exists "$a_directory")"
+}
+
+function test_successful_assert_is_directory() {
+  local a_directory
+  a_directory="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+  assert_empty "$(assert_is_directory "$a_directory")"
+}
+
+function test_unsuccessful_assert_is_directory() {
+  local a_directory="a_random_directory_that_will_not_exist"
+
+  assert_equals\
+    "$(Console::printFailedTest \
+      "Unsuccessful assert is directory" "$a_directory" "to be a directory" "but is not a directory")"\
+    "$(assert_is_directory "$a_directory")"
+}
+
+function test_unsuccessful_assert_is_directory_when_a_file_is_given() {
+  local a_file
+  a_file="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/$(basename "${BASH_SOURCE[0]}")"
+
+  assert_equals\
+    "$(Console::printFailedTest\
+      "Unsuccessful assert is directory when a file is given" "$a_file" "to be a directory" "but is not a directory")"\
+    "$(assert_is_directory "$a_file")"
+}
+
+function test_successful_assert_is_directory_empty() {
+  local a_directory
+  a_directory=$(mktemp -d)
+
+  assert_empty "$(assert_is_directory_empty "$a_directory")"
+
+  rmdir "$a_directory"
+}
+
+function test_unsuccessful_assert_is_directory_empty() {
+  local a_directory
+  a_directory="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+  assert_equals\
+    "$(Console::printFailedTest \
+      "Unsuccessful assert is directory empty" "$a_directory" "to be empty" "but is not empty")"\
+    "$(assert_is_directory_empty "$a_directory")"
+}
+
+function test_successful_assert_is_directory_readable() {
+  local a_directory
+  a_directory=$(mktemp -d)
+
+  assert_empty "$(assert_is_directory_readable "$a_directory")"
+
+  rmdir "$a_directory"
+}
+
+function test_unsuccessful_assert_is_directory_readable_when_a_file_is_given() {
+  local a_file
+  a_file="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/$(basename "${BASH_SOURCE[0]}")"
+
+  assert_equals\
+    "$(Console::printFailedTest\
+      "Unsuccessful assert is directory readable when a file is given" \
+      "$a_file" "to be readable" "but is not readable")"\
+    "$(assert_is_directory_readable "$a_file")"
+}
+
+function test_unsuccessful_assert_is_directory_readable_without_execution_permission() {
+  if [[ "$_OS" == "Windows" ]]; then
+    return
+  fi
```

Comment:
I am ok with this for now. We can investigate why is it not working for Windows in a follow up iteration and check if it's even worth it or even possible to make this test compatible with Windows. CC: @Tito-Kati, @cmayo